cmd/bundle/generate/utils.go

```diff
@@ -0,0 +1,99 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/notebook"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"golang.org/x/sync/errgroup"
+)
+
+type notebookDownloader struct {
+	notebooks map[string]string
+	w         *databricks.WorkspaceClient
+	outputDir string
+}
+
+func (n *notebookDownloader) MarkForDownload(ctx context.Context, task *jobs.Task) error {
+	if task.NotebookTask == nil {
+		return nil
+	}
+
+	info, err := n.w.Workspace.GetStatusByPath(ctx, task.NotebookTask.NotebookPath)
+	if err != nil {
+		return err
+	}
+
+	ext := notebook.GetExtensionByLanguage(info)
+
+	filename := path.Base(task.NotebookTask.NotebookPath) + ext
+	targetPath := filepath.Join(n.outputDir, filename)
+
+	n.notebooks[targetPath] = task.NotebookTask.NotebookPath
+
+	task.NotebookTask.NotebookPath = strings.Join([]string{".", filename}, string(filepath.Separator))
```

Comment:
This means the sources are placed in the same directory as the YAML configuration. If we apply defaults we use in the templates we should make this `../src`, and/or make the directory where we place code configurable as a flag on the command as well (it can default to `./src` if the configuration output dir defaults to `./resources`).